src/twister2/twister_config.py

```diff
@@ -28,6 +29,7 @@ class TwisterConfig:
     selected_platforms: list[str] = field(default_factory=list, repr=False)
     platforms: list[PlatformSpecification] = field(default_factory=list, repr=False)
     hardware_map_list: list[HardwareMap] = field(default_factory=list, repr=False)
+    quarantine: QuarantineProcessor = QuarantineProcessor()
```

Comment:
We should not keep an instance of QuarantineProcessor in TwisterConfiguration. TwisterConfiguration should stay as simple as possible, and should contain only simple data structure like string, int, boolean or list of strings. I know that we keep PlatformSpecification and HardwareMap, but both of them are simple data classes. 
So we should have only these two options: `quarantine_list_path` and `quarantine_verify`, and  `QuarantineProcessor` class should be instantiated in function where we use it for filtering.

Reply:
Thanks for comments. I simplified QuarantineData dataclass and instantiated them in TwisterConfiguration. Please check, if that solution is ok.